File: TaggyAppBackend/TaggyAppBackend.Api/Models/Dtos/Group/GetGroupDto.cs
using Sieve.Attributes;
using TaggyAppBackend.Api.Models.Dtos.GroupUser;
using TaggyAppBackend.Api.Models.Dtos.Tag;
using TaggyAppBackend.Api.Models.Enums;

namespace TaggyAppBackend.Api.Models.Dtos.Group;

public class GetGroupDto : GetDto
{
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public GroupRole CurrentUserRole { get; set; }
    public IEnumerable<GetGroupUserDto> Users { get; set; } = [];
    public IEnumerable<GetTagDto> Tags { get; set; } = [];
 }

File: TaggyAppFrontend/src/app/models/dtos/group/getGroupDto.ts
import { GroupRole } from '../../enums/groupRole';
import { GetDto } from '../getDto';
import { GetGroupUserDto } from '../groupUser/getGroupUserDto';
import { GetTagDto } from '../tag/getTagDto';

export class GetGroupDto extends GetDto {
  name!: string;
  description?: string;
  currentUserRole!: GroupRole;
  users!: GetGroupUserDto[];
  tags!: GetTagDto[];
}
